syntax = "proto3";

import "google/protobuf/empty.proto";


service GameService {
  rpc sayHello (google.protobuf.Empty) returns (StringRequest);
  rpc cadastrarCliente (StringRequest) returns (ClienteResponse);
  rpc acessarSala (SalaRequest) returns (SalaResponse);
  rpc verMao (StringRequest) returns (CartasResponse);
  rpc pegarCartaDeck (StringRequest) returns (CartasResponse);
  rpc desistirJogo (StringRequest) returns (boolResponse);
}


message StringRequest {
  string string = 1;
}

message SalaRequest {
  string token = 1;
  string name = 2;
}


message SalaResponse {
  bool finalizada = 1;
  string message = 2;
  bool erro = 3;
}

message CartasResponse {
  repeated string cartas = 1;
}

message ClienteResponse {
  string name = 1;
  string chave = 2;
}

message boolResponse{
  bool resp = 1;
}